src/machine_learning/optimization/k_nearest_neighbors.rs

```diff
@@ -0,0 +1,294 @@
+#![allow(dead_code)]
+#![allow(unused_variables)]
+
+use std::{
+    cmp,
+    collections::{BTreeMap, HashMap},
+};
+
+use crate::math::{abs, square_root};
+#[derive(Debug, Clone)]
+pub struct Point {
+    x: f64,        // x-axis
+    y: f64,        // y-axis
+    label: String, // A label denotes the classification of the data. For instance, 'Football' might be labeled as 'Sport'
+}
+
+impl Point {
+    pub fn build(x: f64, y: f64, label: String) -> Self {
+        Self { x, y, label }
+    }
+}
+
+pub enum DistanceMeasurementFormula {
+    EucledianDistance, // The Euclidean Distance Formula will be employed to calculate the distance between two points
+}
+
+impl DistanceMeasurementFormula {
+    fn distance(&self, source_point: &Point, destination_point: &Point) -> f64 {
+        match self {
+            Self::EucledianDistance => Self::eucledian_distance(source_point, destination_point),
+        }
+    }
+
+    fn eucledian_distance(source_point: &Point, destination_point: &Point) -> f64 {
+        let distance = square_root(
+            (destination_point.x - source_point.x) * (destination_point.x - source_point.x)
+                + (destination_point.y - source_point.y) * (destination_point.y - source_point.y),
+        );
+        abs(distance)
+    }
+}
```

Comment:
Why not to use [existing implementation](https://github.com/TheAlgorithms/Rust/blob/c2009cd31ea0d5171ce1c12b4c3742a2676cd07e/src/geometry/point.rs#L25)?

Reply:
We can. However , i have changed my code to use n-dimensional data. The existing implementation is only for 2D data.